app/auth/register.tsx

```diff
@@ -0,0 +1,286 @@
+                }}
+              >
+                <Picker.Item label="Select Gender" value="" enabled={false} />
+                <Picker.Item label="Male" value="Male" />
+                <Picker.Item label="Female" value="Female" />
+              </Picker>
+            </View>
+          </View>
+          <Text className="text-error-500 my-2">{formik.errors.gender}</Text>
+        </View>
+
+        <View>
+          <DatePicker
+            placeholder="Date of Birth"
+            onSubmit={(date: Date) => {
+              formik.setFieldValue('dob', date);
+            }}
+          />
+          <Text className="text-error-500 my-2">{formik.errors.dob}</Text>
+        </View>
+
+        <View>
+          <View
+            className={`flex-row items-center px-3 py-2.5 rounded-lg border border-gray-300 shadow shadow-gray-50 bg-primary-light`}
+          >
+            <Ionicons name="person" size={20} color="gray" className="mr-2" />
+
+            <TextInput
+              testID="password"
+              placeholder="Password"
+              onChangeText={formik.handleChange('password')}
+              value={formik.values.password}
+              style={{ flex: 1 }}
+              autoCapitalize="none"
+              className={`ml-2`}
+              secureTextEntry={!showPassword}
+            />
+            <Ionicons
+              name={showPassword ? 'eye-off' : 'eye'}
+              size={20}
+              color="gray"
+              className="mx-2"
+              onPress={() => {
+                setShowPassword(!showPassword);
+              }}
+            />
+          </View>
+          <Text className="text-error-500 my-2">{formik.errors.password}</Text>
+        </View>
+
+        <View className="flex flex-col gap-4 mt-2">
+          <View className="mt-2 flex-row ml-2">
+            <Checkbox
+              className="w-4 h-4 rounded border border-gray-300"
+              value={TCAccepted}
+              onValueChange={setTCAccepted}
+              color={TCAccepted ? '#8667F2' : undefined}
+            />
+            <Text className="ml-2">
+              I agree to <Text className="text-action-600">the Terms & Conditions</Text> of Pulse
+            </Text>
+          </View>
+          <TouchableOpacity
+            onPress={() => {
+              if (!TCAccepted) {
+                Alert.alert('Warning', 'Please accept terms and conditions');
+                return;
+              }
+              formik.handleSubmit();
+            }}
+            className="bg-action-500 p-4 rounded-lg items-center mt-4"
+          >
+            {loading ? (
+              <ActivityIndicator color="#fff" />
+            ) : (
+              <Text className={`text-white text-lg`}>Register</Text>
+            )}
+          </TouchableOpacity>
+        </View>
+        <View className="mt-6">
+          <Text>
+            Already have an account? <Text className="text-action-500">Sign in</Text>
```

Comment:
```suggestion
            Already have an account? <Text className="text-action-500">Sign in</Text>
```
Here the user will be able to switch to sign in screen after pressing on sign in. if already have an account.

Reply:
Thank you for this, now it's good to go